ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/bonsai/worldview/BonsaiWorldState.java

```diff
@@ -337,28 +341,34 @@ private void clearStorage(
       try {
         final StorageConsumingMap<StorageSlotKey, DiffBasedValue<UInt256>> storageToDelete =
             worldStateUpdater.getStorageToUpdate().get(address);
-        Map<Bytes32, Bytes> entriesToDelete = storageTrie.entriesFrom(Bytes32.ZERO, 256);
-        while (!entriesToDelete.isEmpty()) {
-          entriesToDelete.forEach(
-              (k, v) -> {
-                final StorageSlotKey storageSlotKey =
-                    new StorageSlotKey(Hash.wrap(k), Optional.empty());
-                final UInt256 slotValue = UInt256.fromBytes(Bytes32.leftPad(RLP.decodeValue(v)));
-                maybeStateUpdater.ifPresent(
-                    bonsaiUpdater ->
-                        bonsaiUpdater.removeStorageValueBySlotHash(
-                            address.addressHash(), storageSlotKey.getSlotHash()));
-                storageToDelete
-                    .computeIfAbsent(
-                        storageSlotKey, key -> new DiffBasedValue<>(slotValue, null, true))
-                    .setPrior(slotValue);
-              });
-          entriesToDelete.keySet().forEach(storageTrie::remove);
-          if (entriesToDelete.size() == 256) {
-            entriesToDelete = storageTrie.entriesFrom(Bytes32.ZERO, 256);
-          } else {
-            break;
+        if (storageToDelete != null) {
```

Comment:
did you tried the proposition I shared on discord https://discord.com/channels/905194001349627914/1261348506648707133/1296138719115345980

regarding my comment here https://discord.com/channels/905194001349627914/1261348506648707133/1294238512157032491

Reply:
Yes I did try it, and I think I then got side tracked by hitting some unrelated issues. So I've reinstated your proposed implementation of `clearStorage()` in the latest commits.